### PR TITLE
Add Nomenclature / Cheatsheet page

### PR DESCRIPTION
**Pull request by diesalbla**
Adds a page with the types and names of the many functions and methods of `cats`. 
The types are written in a Haskell-lke form, as separate currified type signatures, without the names of the parameters.

### CODE REVIEW

**Comment by arosien** [comment]
This is great!

Perhaps "Nomenclature" isn't the right term, it's more of an overview, or cheatsheet, or index of the major functions?

(disclaimer: I made something similar, as a PDF, at https://arosien.github.io/cats-cheatsheets/typeclasses.pdf)

**Comment by diesalbla** [comment]
@arosien Well, I am really sorry to say, I did not know of your cheatsheet. Otherwise, I would have saved me some time by `sed`-ing it :(.  I will add a link to it. Of course, Markdown is not quite as nice as `fourier`, but still, that be good enough.

**Comment by diesalbla** [comment]
I am putting this to submission, and removing the WIP label. While this page may be further refined, the current version is wide enough for a first version.

File: docs/src/main/tut/nomenclature.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+This page contains a catalogue of the major functions, type classes, and types aliases within the `cats` library. This catalogue serves, on one hand, as a bird's-eye view of the capabilities of each class. It is also intended as a go-to page for any `cats`-using developer, who may not remember the answer to any of the questions above.
+
+Those looking for a printable version may want to check out the [cats-cheatsheet](https://arosien.github.io/cats-cheatsheets/typeclasses.pdf) file.
+
+#### Simplifications
+
+Because `cats` is a library for Scala, and because Scala has many more knobs and switches, the actual definitions and implementations of some functions in the library can seem a bit too obfuscated at first sight. To alleviate this, in this glossary we focus on the plain type signatures of the method, and ignore many of the details from Scala. In particular, in our type signatures:
+
+- We use `A,B,C` for type variables of kind `*`, and `F, G, H` for type variables of a higuer kind.
```
Comment (kailuowang) [inline docs/src/main/tut/nomenclature.md]: higuer -> higher

**Pull request merged**